src/worker-type.js

```diff
@@ -414,14 +414,15 @@ WorkerType.prototype.createLaunchSpec = function(bid) {
  * function will throw if there is an error found or will return
  * a dictionary of all the launch specs!
  */
-WorkerType.prototype.testLaunchSpecs = function() {
+WorkerType.prototype.testLaunchSpecs = function(awsManager) {
```

Comment:
I'd rather that we don't pass in `awsManager`.  This function is about whether or not the worker type config is structurally valid and not so much about whether those launch specs actually can or should be launchable.  Those checks should be done in `workerTypeCanLaunch` in the aws manager.  For this function, let's instead just assume that the list of available availability zones are those in the configuration, and if none are defined, just do it by region and instance type.  

If we absolutely have to have at least a single availability zone per region for the code to execute, let's instead just assume `${region}-a` with no per-az config.  Because of how availability zone naming works, I'm pretty sure that the 'a' zone is guaranteed to always exist.  Either way, since these launchSpecs are never actually used to create real instances, we should be fine to use a hypothetical availability zone.